src/test/java/com/iggroup/universityworkshopmw/domain/services/OpenPositionsServiceTest.java

```diff
@@ -140,123 +166,147 @@ public void shouldCloseSpecifiedPosition() throws Exception {
+   @Test
+   public void updateMarketPrice_shouldUpdateClientRunningProfitAndLossAndAvailableFunds() throws Exception {
+      initialiseClientPositions();
+
+      openPositionsService.updateMarketPrice("market_1", 200.00);
+
+      ArgumentCaptor<String> clientIdCaptor = ArgumentCaptor.forClass(String.class);
+      ArgumentCaptor<Double> sumProfitAndLossCaptor = ArgumentCaptor.forClass(Double.class);
+      verify(clientService, times(2)).updateRunningProfitAndLoss(clientIdCaptor.capture(), sumProfitAndLossCaptor.capture());
+      assertThat(clientIdCaptor.getAllValues()).containsOnly("client_1", "client_3");
+      assertThat(sumProfitAndLossCaptor.getAllValues()).containsOnly(1324.12, 18950.0);
    }
 
    private void initialiseOpenPositions() {
       openPosition1 = OpenPosition.builder()
-         .id("pos_1")
-         .marketId("market_1")
-         .profitAndLoss(1234.55)
-         .openingPrice(100.00)
-         .buySize(12)
-         .build();
+            .id("pos_1")
+            .marketId("market_1")
+            .profitAndLoss(1234.55)
+            .openingPrice(100.00)
+            .buySize(12)
+            .build();
 
       openPosition2 = OpenPosition.builder()
-         .id("pos_2")
-         .marketId("market_2")
-         .profitAndLoss(124.12)
-         .openingPrice(140.00)
-         .buySize(1)
-         .build();
+            .id("pos_2")
+            .marketId("market_2")
+            .profitAndLoss(124.12)
+            .openingPrice(140.00)
+            .buySize(1)
+            .build();
 
       openPosition3 = OpenPosition.builder()
-         .id("pos_3")
-         .marketId("market_3")
-         .profitAndLoss(12.1)
-         .openingPrice(50.5)
-         .buySize(70)
-         .build();
+            .id("pos_3")
+            .marketId("market_3")
+            .profitAndLoss(12.1)
+            .openingPrice(50.5)
+            .buySize(70)
+            .build();
 
       openPosition4 = OpenPosition.builder()
-         .id("pos_4")
-         .marketId("market_1")
-         .profitAndLoss(134.00)
-         .openingPrice(10.5)
-         .buySize(100)
-         .build();
+            .id("pos_4")
+            .marketId("market_1")
+            .profitAndLoss(134.00)
+            .openingPrice(10.5)
+            .buySize(100)
+            .build();
 
       openPosition5 = OpenPosition.builder()
-         .id("pos_5")
-         .marketId("market_4")
-         .profitAndLoss(543.98)
-         .openingPrice(234.42)
-         .buySize(50)
-         .build();
+            .id("pos_5")
+            .marketId("market_4")
+            .profitAndLoss(543.98)
+            .openingPrice(234.42)
+            .buySize(50)
+            .build();
 
       openPosition6 = OpenPosition.builder()
-         .id("pos_6")
-         .marketId("market_1")
-         .profitAndLoss(12.00)
-         .openingPrice(250.00)
-         .buySize(12)
-         .build();
+            .id("pos_6")
```

Comment:
I think we might have different spacing settings on Intellij!

Reply:
aha oops! What is yours? I will change it to be the same